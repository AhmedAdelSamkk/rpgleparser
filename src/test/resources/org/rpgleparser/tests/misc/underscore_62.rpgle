     C                   EVAL     getFileByName('SROORSHE')
     C                   EVAL     _getFileByName('SROORSHE')

      /free
         _getFileByName('SROORSHE'); 
      /end-free 
                 
