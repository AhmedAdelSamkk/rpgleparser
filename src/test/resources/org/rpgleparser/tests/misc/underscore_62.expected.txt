/*===TOKENS===*/
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_EVAL                  <EVAL>
ID                       <getFileByName>
'('                      <(>
StringLiteralStart       <'>
StringContent            <SROORSHE>
StringLiteralEnd         <'>
')'                      <)>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_EVAL                  <EVAL>
ID                       <_getFileByName>
'('                      <(>
StringLiteralStart       <'>
StringContent            <SROORSHE>
StringLiteralEnd         <'>
')'                      <)>
C_FREE_NEWLINE           <
>
DIRECTIVE                < />
DIR_FREE                 <free>
EOL                      <>
ID                       <_getFileByName>
'('                      <(>
StringLiteralStart       <'>
StringContent            <SROORSHE>
StringLiteralEnd         <'>
')'                      <)>
FREE_SEMI                <;>
DIRECTIVE                < />
DIR_ENDFREE              <end-free>
EOL                      <>
/*===TREE===*/
(r
  (statement
    (cspec_fixed
      C
      (cs_controlLevel '  ')
      (onOffIndicatorsFlag ' ')
      (cs_indicators '  ')
      (factor '              ')
      (cspec_fixed_standard
        (csEVAL
          EVAL
          (c_free
            (expression
              (function
                (functionName (free_identifier (idOrKeyword getFileByName)))
                (args ( (expression (literal ' SROORSHE ')) ))
              )
            )
          )
          \n
        )
      )
    )
  )
  (statement
    (cspec_fixed
      C
      (cs_controlLevel '  ')
      (onOffIndicatorsFlag ' ')
      (cs_indicators '  ')
      (factor '              ')
      (cspec_fixed_standard
        (csEVAL
          EVAL
          (c_free
            (expression
              (function
                (functionName (free_identifier (idOrKeyword _getFileByName)))
                (args ( (expression (literal ' SROORSHE ')) ))
              )
            )
          )
          \n
        )
      )
    )
  )
  (statement (directive ' /' (beginfree_directive free) ))
  (statement
    (free
      (baseExpression
        (op
          (op_callp
            (identifier (free_identifier (idOrKeyword _getFileByName)))
            (
            (expression (literal ' SROORSHE '))
            )
          )
        )
      )
      ;
    )
  )
  (statement (directive ' /' (endfree_directive end-free) ))
)
/*======*/